src/checkpointing/checkpoint.py

```diff
@@ -0,0 +1,29 @@
+import sys
+from datetime import datetime
+
+class Checkpoint:
+    def __init__(self):
+        self._file = "./assets/checkpoint.txt"
+
+    # This function will write to the checkpoint file
+    # if there is an existing checkpoint it will overwrite that checkpoint
+    # date_to_checkpoint : Will be the date of the last row processed, this should be a string from a row entry
+    # end_date: Will be the ending date for the range of entries being queried, this should be a string from a row entry
+    def write_to_file(self, date_to_checkpoint, end_date):
+        with open(self._file, "w") as fstream:
+            fstream.write(date_to_checkpoint + "," + end_date)
+
+    # This function will read from the checkpoint file, it will get the last date that was checkpointed
+    def read_from_file(self):
+        with open(self._file, "r") as fstream:
+            for line in fstream:
+                dates = line.split(",")
```

Comment:
You're only getting the last dates right? You shouldn't have to iterate through every single line like this.

Reply:
iirc, the file rewrites the whole file on creation so there is only one line to read.

Reply:
It's just one line

Reply:
Ah ok.